internal/server/singleprocess/service_url.go

```diff
@@ -40,5 +50,9 @@ func (s *service) initURLGuestAccount(acceptURLTerms bool) error {
 	}
 
 	s.urlConfig.APIToken = accountResp.Token
+	if err := s.state.ServerURLTokenSet(accountResp.Token); err != nil {
```

Comment:
Responding to your comment about setting the token in the success case...I should be doing it here for the initial check at the beginning of this function to work... is that what you meant @mitchellh ?

Reply:
I don't think so.

I think you just need to put:

> s.urlConfig.APIToken = accountResp.Token

where my original comment is. I think all this here is fine.

Reply:
Oh right, I see what you mean now! :sweat_smile: :+1: